pkg/usage/send.go

```diff
@@ -0,0 +1,100 @@
+package usage
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+)
+
+// Send the analytic results to server
+// if wait is not set then Send will execute go routine
+func (u *Usage) Send(wait bool) error {
+	var fn func() error
+
+	if u.url == "" {
+		fn = u.SendGoogleAnalytic
```

Comment:
If it is sending to google analytics - let it always be unreliable. attempt to send but don't wait or error on failures. 

In case URL based (reliable service),  then we probably need to do a few things:
- during the initialization, check if a connection can be established and messages sent (this needs to be designed) 
- probably a handshake message needs to be established between the client and server on what parameters needs to be sent
- take additional parameters - to check if sending events should be attempted in the background, handle cases like pod restarts, and have the ability to resend events. 

Give all the above, wondering if we should limit this PR to refactoring changes only? Once we finalize the proposed design for sending events reliable service (or call home), we can add the PRs.

Reply:
Looks like this file has been removed from file changes @ `4b00533: Removing sending analytic to REST endpoint`, 
I couldn't find it under the `Files changed` tab, if this has been moved to upcoming PRs, please also update the PR heading, just as the PR description has been updated.